typescript/src/lib/bitcoin/address.ts

```diff
@@ -0,0 +1,121 @@
+import { address as btcjsaddress, payments } from "bitcoinjs-lib"
+import { Hex } from "../utils"
+import { BitcoinNetwork, toBitcoinJsLibNetwork } from "./network"
+
+/**
+ * Creates the Bitcoin address from the public key. Supports SegWit (P2WPKH) and
+ * Legacy (P2PKH) formats.
+ * @param publicKey - Public key used to derive the Bitcoin address.
+ * @param bitcoinNetwork - Target Bitcoin network.
+ * @param witness - Flag to determine address format: true for SegWit (P2WPKH)
+ *        and false for Legacy (P2PKH). Default is true.
+ * @returns The derived Bitcoin address.
+ */
+export function publicKeyToAddress(
+  publicKey: Hex,
+  bitcoinNetwork: BitcoinNetwork,
+  witness: boolean = true
+): string {
+  const network = toBitcoinJsLibNetwork(bitcoinNetwork)
+
+  if (witness) {
+    // P2WPKH (SegWit)
+    return payments.p2wpkh({ pubkey: publicKey.toBuffer(), network }).address!
+  } else {
+    // P2PKH (Legacy)
+    return payments.p2pkh({ pubkey: publicKey.toBuffer(), network }).address!
+  }
+}
+
+/**
+ * Converts a public key hash into a P2PKH/P2WPKH address.
+ * @param publicKeyHash Public key hash that will be encoded. Must be an
+ *        unprefixed hex string (without 0x prefix).
+ * @param witness If true, a witness public key hash will be encoded and
+ *        P2WPKH address will be returned. Returns P2PKH address otherwise
+ * @param bitcoinNetwork Network the address should be encoded for.
+ * @returns P2PKH or P2WPKH address encoded from the given public key hash
+ * @throws Throws an error if network is not supported.
+ */
+function publicKeyHashToAddress(
+  publicKeyHash: string,
```

Comment:
I agree, we should choose one option: either `string` or `Hex` to represent hexadecimal values.
The `Hex` would be better, because of validation.
However, can we use `Hex` for exported functions?

Reply:
Yeah we should use `Hex` and there is ongoing work supposed to address that: https://github.com/keep-network/tbtc-v2/issues/713